Source/scrollrt.cpp

```diff
@@ -1192,28 +1159,28 @@ void DrawGame(const Surface &fullOut, int x, int y)
 		break;
 	case SDIR_SW:
 		sx -= TILE_WIDTH;
-		ShiftGrid(&x, &y, -1, 0);
+		position += DIR_W;
 		columns++;
 		rows++;
 		break;
 	case SDIR_W:
 		sx -= TILE_WIDTH;
-		ShiftGrid(&x, &y, -1, 0);
+		position += DIR_W;
 		columns++;
 		break;
 	case SDIR_NW:
 		sx -= TILE_WIDTH / 2;
 		sy -= TILE_HEIGHT / 2;
-		x--;
+		position += DIR_NW;

```

Comment:
Nice now I finally understand this code, it need to shift if moving N or W :D

Reply:
At least someone understands it. I don't.